cpg_infra/utils.py

```diff
@@ -1,16 +1,35 @@
 """Utility methods that may be useful for multiple projects"""
-import os
 import contextlib
+import os
+from typing import Any, Callable
 
 import pulumi
 
 DEFAULT_ALLOWED_EXTENSIONS = frozenset({'.py', '.txt', '.json'})
 
 
 def archive_folder(
-    path: str, allowed_extensions: frozenset[str] = DEFAULT_ALLOWED_EXTENSIONS
+    path: str,
+    allowed_extensions: frozenset[str] = DEFAULT_ALLOWED_EXTENSIONS,
+    apply_fun_map: dict[str, tuple[Callable, Any]] | None = None,
```

Comment:
Instead of functions to override a specific file's behaviour, add an ignore list, and a list of additional files you want to add.